src/attack.c

```diff
@@ -890,8 +888,20 @@ apply_source_iface_preference (void)
 
       g_free (ipstr);
       g_free (ip6str);
-      return 0;
+      ret = 0;
     }
+
+  if (msg)
```

Comment:
```suggestion
  if (ret < 0 && msg)
```
Maybe also use the ret value here.

Reply:
Not necessary. It would be redundant, as msg will be != NULL, if and only if ret is less than 0.